### PR TITLE
migrations: make errors schema match SaaS

### PR DESCRIPTION
**Pull request by dbanda**
This makes the `errors_local` and `errors_dist_ro` table schemas match what is in SaaS. 

It adds two new migration:
1. Adds indexes and updates the table settings to match SaaS
2. Adds the hierarchical_hashes and tags_hash_map fields to match the SaaS schema.

### CODE REVIEW

File: snuba/snuba_migrations/events/0018_errors_ro_add_tags_hash_map.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = True
+
+    def forwards_ops(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.EVENTS,
```
Comment (MeredithAnya) [inline snuba/snuba_migrations/events/0018_errors_ro_add_tags_hash_map.py:16]: should these StorageSetKey's be `StorageSetKey.EVENTS_RO` instead?

Comment (dbanda) [reply snuba/snuba_migrations/events/0018_errors_ro_add_tags_hash_map.py]: fixed

**Review by MeredithAnya — APPROVED** [review]

**Comment by nikhars** [comment]
Are there any discrepancies in the errors_dist table? I am wondering how come the new migrations you are adding on `errors_dist_ro`, not needed on `errors_dist`

**Comment by dbanda** [comment]
> Are there any discrepancies in the errors_dist table? I am wondering how come the new migrations you are adding on `errors_dist_ro`, not needed on `errors_dist`

The `errors_dist` table has no discrepancies.

**Pull request merged**